src/main/java/model/Rank.java

```diff
@@ -0,0 +1,43 @@
+package model;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public enum Rank {
+    FIRST(6, 2000000000),
+    SECOND(5, 30000000),
+    THIRD(5, 1500000),
+    FOUTH(4, 50000),
+    FIFTH(3, 5000),
+    NONE(0, 0);
+
+    private final int matchCount;
+    private final int winningMoney;
+
+    Rank(int matchCount, int winningMoney) {
+        this.matchCount = matchCount;
+        this.winningMoney = winningMoney;
+    }
+
+    public int getMatchCount() {
+        return matchCount;
+    }
+
+    public int getWinningMoney() {
+        return winningMoney;
+    }
+
+    public static Rank getRankByMatchCount(int matchCount, boolean hasBonusNo) {
+        List<Rank> ranks = Arrays.stream(Rank.values())
+                .collect(Collectors.toList());
+        if (!hasBonusNo) {
```

Comment:
제가 알기로는 ``로또 숫자가 5개 일치하고 BonusNumber가 일치하는 경우`` **2등**인 것으로 알고 있습니다.  
테스트코드도 fail이 뜨고 있는 것으로 보아, ``!``을 제거해야 할 것 같습니다. 
```suggestion
        if (hasBonusNo) {
```

Reply:
보너스 번호의 유무가 영향을 미치는 건 2위와 3위를 결정할때 뿐입니다. 따라서, 보너스 번호를 갖고 있지 않다면 Rank 리스트에서 2위를 제거하는 방향으로 로직을 구현했습니다.